src/gui/macOS/fileprovidersocketcontroller.cpp

```diff
@@ -237,6 +241,26 @@ void FileProviderSocketController::sendAccountDetails() const
     sendMessage(message);
 }
 
+void FileProviderSocketController::sendIgnoreList() const
+{
+    if (!_accountState) {
+        qCWarning(lcFileProviderSocketController) << "No account state available to send ignore list, stopping";
+        return;
+    }
+
+    ExcludedFiles ignoreList;
+    ConfigFile::setupDefaultExcludeFilePaths(ignoreList);
+    ignoreList.reloadExcludeFiles();
+    const auto patterns = ignoreList.activeExcludePatterns();
+    if (patterns.isEmpty()) {
+        qCWarning(lcFileProviderSocketController) << "No active ignore list patterns, not sending.";
+        return;
+    }
+    // Try to come up with a separator that won't clash. I am hoping this is it
+    const auto message = QString(QStringLiteral("IGNORE_LIST:") + patterns.join("_~IL$~_"));
```

Comment:
another possibility might be to serialise the ignore list as a JSON array over the socket -- that should hopefully minimise the risk of hitting some edge cases ;-)